uptrain/framework/checks.py

```diff
@@ -225,3 +227,68 @@ def serialize(self, fpath: t.Optional[str] = None):
 
         with open(fpath, "w") as f:
             jsondump(self.dict(), f)
+
+
+class ExperimentArgs(BaseModel):
+    """Container for arguments to run a LLM experiment. This is the entrypoint to run prompt/model tests with Uptrain for users.
+    This generates multiple prompts, generate LLM responses for them and parses the response to add multiple columns.
+
+    Attributes:
+        prompt_template (str): Prompt template to generate multiple prompts to experiment with.
+        prompt_params (dict(str, str)): Dictionary of prompt variables with experiment with.
+        models (str or list[str]): Either a single model or list of models to experiment with.
+        context_vars (dict or list): List of dataset variables with mapping between variable name in prompt and variable name in dataset
+        temperature (float): Temperature for the LLM response generation.
+        col_out_response (str): LLM output will be saved under this column name
+        col_out_mapping (dict): Mapping used to parse LLM response
+    """
+
+    prompt_template: str
+    prompt_params: dict = {}
+    models: t.Union[str, list[str]]
+    context_vars: t.Union[dict, list]
+    temperature: float = 1.0
+    col_out_response: str = 'response'
+    col_out_mapping: t.Optional[dict] = None
+
+    def get_preprocessors(self):
```

Comment:
if users dont use these methods, can we add an underscore prefix?

Reply:
Done